tests/reference/tests/chocolateysource/apply_manifest_without_location.rb

```diff
@@ -0,0 +1,24 @@
+require 'chocolatey_helper'
+test_name 'MODULES-3037 - Add Source: Apply manifest without location'
+confine(:to, :platform => 'windows')
+
+backup_config
+
+# arrange
+chocolatey_src = <<-PP
+  chocolateysource {'chocolatey':
+    ensure   => present,
+  }
+PP
+
+# teardown
+teardown do
+  reset_config
+end
+
+# act
+step 'Apply Manifest'
+apply_manifest(chocolatey_src, :catch_failures => true) do
+  step 'Verify Result'
+  assert_match(/Notice: \/Stage\[main\]\/Main\/Chocolateysource\[chocolatey\]\/location: location changed 'https:\/\/chocolatey.org\/api\/v2\/' to 'chocolatey'/, stdout, "stdout did not match expected")
```

Comment:
Why would someone define a resource as `chocolateysource { 'foo': }` like that? There are other T&P I could make the same argument on. :D

And to remove foo, I agree that's how you do it, nobody said it would be done any other way (unless I missed something).


And the second, if you wanted to have  a useless name that didn't define the location very well - to put it bluntly, that's on you.